tests/test_inspect_pkg.py

```diff
@@ -207,14 +207,22 @@ def test_which_package(tmp_path: Path):
 @pytest.mark.benchmark
 def test_which_package_battery(tmp_path: Path):
     # regression: https://github.com/conda/conda-build/issues/5126
+
+    # NOTE: CodSpeed on Python 3.12+ activates the stack profiler trampoline backend
+    # and thus runs the test twice (once without profiling and once with profiling),
+    # unfortunately this means that on the second iteration tmp_path is no longer empty
+    # so we create a randomized unique directory to compensate
+    tmp_path = tmp_path / uuid4().hex
+    tmp_path.mkdir()
+
     # create a dummy environment
     (tmp_path / "conda-meta").mkdir()
     (tmp_path / "conda-meta" / "history").touch()
     (tmp_path / "lib").mkdir()
 
     # dummy packages with files
     removed = []
-    for _ in range(100):
+    for _ in range(10):
```

Comment:
Since CodSpeed is a profiler things naturally run slower, we change this value in order to get a more reasonable run time (30min+ vs seconds)